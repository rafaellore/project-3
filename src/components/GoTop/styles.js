import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 2.5rem;
    height: 2.5rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    border-radius: 50%;
  `}
`;
