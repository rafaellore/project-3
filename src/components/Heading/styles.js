import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, isLightText }) => css`
    color: ${isLightText ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
