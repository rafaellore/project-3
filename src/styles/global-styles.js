// global styles is like a css reset
// we can use theme here too

import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  body {
    font-size: 1.6rem;
    font-family: 'Open Sans', sans-serif;
  }
  h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat', sans-serif;
  }
`;
