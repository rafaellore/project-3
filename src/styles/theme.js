// here we set some constants related to styles, like fonts, colors etc
// this const is used on ThemeProvider of styled-components with allows you to use in the childrens styles

export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },

  spacings: {
    xsmall: '0.5rem',
    small: '1rem',
    medium: '1.5rem',
    large: '2rem',
    xlarge: '2.5rem',
    xxlarge: '3rem',
    huge: '3.5rem',
    xhuge: '4rem',
  },

  font: {
    sizes: {
      xsmall: '0.5rem',
      small: '1rem',
      medium: '1.5rem',
      large: '2rem',
      xlarge: '2.5rem',
      xxlarge: '3rem',
      huge: '3.5rem',
      xhuge: '4rem',
    },
  },

  media: {
    lteMedium: '(max-width: 768px)',
  },
};
