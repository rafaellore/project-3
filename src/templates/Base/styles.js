import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 54px;

    @media ${theme.media.lteMedium} {
      padding-top: 0;
    }
  `}
`;
