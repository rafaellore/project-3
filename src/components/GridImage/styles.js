import styled, { css } from 'styled-components';

import { Container as Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
