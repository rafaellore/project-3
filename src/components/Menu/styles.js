import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.white};
    transition: all 300ms ease-in-out;

    // to import styles from another place and edit himself
    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    // to import styles from another place and edit himself
    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    //media querie
    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible()}

      //styles from another place inside media querie
      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.large};
        margin: 0 auto;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    //media querie
    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 2.5rem;
    height: 2.5rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    display: none;
    opacity: 0;
    visibility: hidden;

    // needs this to work
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;

      opacity: initial;
      visibility: visible;
    }

    > svg {
      width: 1.5rem;
      height: 1.5rem;
    }
  `}
`;
