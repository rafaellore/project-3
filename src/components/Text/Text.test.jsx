import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Text } from '.';
import { theme } from '../../styles/theme';

describe('<Text />', () => {
  it('should render', () => {
    renderTheme(<Text>Simple text</Text>);
  });

  it('renders the p tag correctly', () => {
    const { container } = renderTheme(<Text>Simple text</Text>);

    const p = container.querySelector('p');

    expect(p.tagName.toLowerCase()).toBe('p');
  });

  it('renders the text currectly', () => {
    renderTheme(<Text>Simple text</Text>);

    expect(screen.getByText('Simple text')).toBeInTheDocument();
  });

  it('renders with the currect size', () => {
    renderTheme(<Text>Simple text</Text>);

    const paragraph = screen.getByText('Simple text');

    expect(paragraph).toHaveStyle({
      'font-size': theme.font.sizes.medium,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Text>Simple text</Text>);

    // with firstChild this gets only the p tag
    expect(container.firstChild).toMatchSnapshot();
  });
});
