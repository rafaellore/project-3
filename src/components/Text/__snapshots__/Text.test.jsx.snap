// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match snapshot 1`] = `
.c0 {
  font-size: 1.5rem;
}

<div
  class="c0"
>
  Simple text
</div>
`;
